core/new-gui/src/app/workspace/component/workflow-editor/workflow-editor.component.spec.ts

```diff
@@ -859,5 +872,57 @@ describe("WorkflowEditorComponent", () => {
       expect(jointGraphWrapper.getCurrentHighlightedOperatorIDs()).toContain(mockScanPredicate.operatorID);
       expect(jointGraphWrapper.getCurrentHighlightedOperatorIDs()).toContain(mockResultPredicate.operatorID);
     });
+
+    //undo
+    it("should undo action when press command + Z or control + Z", () => {
+      spyOn(workflowVersionService, "getDisplayParticularVersionStream").and.returnValue(of(false));
+      spyOn(workflowCollabService, "isLockGranted").and.returnValue(true);
+      spyOn(undoRedoService, "canUndo").and.returnValue(true);
+      let undoSpy = spyOn(undoRedoService, "undoAction");
+      fixture.detectChanges();
+      const commandZEvent = new KeyboardEvent("keydown", { key: "Z", metaKey: true, shiftKey: false });
+      (document.activeElement as HTMLElement)?.blur();
+      document.dispatchEvent(commandZEvent);
+      fixture.detectChanges();
+      expect(undoSpy).toHaveBeenCalled();
```

Comment:
I see two `expect(undoSpy).toHaveBeenCalled()` here. If I understand correctly, since the first event `metaKey+Z` can call `undoAction`, it does not matter the second event `ctrl+Z` is called or not, this check will pass anyway. So the second `expect(undoSpy).toHaveBeenCalled()` is not valid. Maybe it's better to write them as two separate tests? Feel free to correct me if I am wrong.

Reply:
@Xiao-zhen-Liu your comment is correct, we can use the `toHaveBeenCalledTimes` in the second time to verify it's called twice. Maybe @Xiao-zhen-Liu can make this simple change.